tests/rptest/tests/redpanda_startup_test.py

```diff
@@ -96,3 +99,93 @@ def check_fips_mode_metric(
+        metric responds appropriately
+
+        1. Find the lader
+        2. On a non-leader node, set to 'disabled'
+        3. Restart that node
+        4. Wait for count to go to 1
+        5. On same node, set back to permissive
+        6. Restart node
+        7. Wait for count to go back to 0
+        """
+
+        leader = self.redpanda.controller()
+        non_leader_nodes = list(set(self.redpanda.nodes).difference({leader}))
+        assert len(non_leader_nodes) > 0, "Could not find any non-leader nodes"
+        target_node = non_leader_nodes[0]
+        self.logger.debug(f'Leader: {leader}')
+        self.logger.debug(f'Non-leader: {target_node}')
+        NON_FIPS_CONFIG = dict(fips_mode="disabled",
+                               openssl_config_file=None,
+                               openssl_module_directory=None)
+        FIPS_CONFIG = dict(
+            fips_mode="permissive",
+            openssl_config_file=self.redpanda.get_openssl_config_file_path(),
+            openssl_module_directory=self.redpanda.
+            get_openssl_modules_directory())
+
+        def get_metrics_value(metrics_name: str,
+                              metrics_endpoint: MetricsEndpoint,
+                              leader_node: ClusterNode) -> int:
+            metrics = self.redpanda.metrics_sample(
+                sample_pattern=metrics_name, metrics_endpoint=metrics_endpoint)
+            assert isinstance(
+                metrics,
+                MetricSamples), f'Failed to get metrics for {metrics_name}'
+
+            samples = [
+                sample for sample in metrics.samples
+                if sample.node == leader_node
+            ]
+            assert len(
+                samples) == 1, f'Invalid number of samples: {len(samples)}'
+            return int(samples[0].value)
+
+        wait_until(
+            lambda: get_metrics_value(
+                metrics_name='redpanda_cluster_non_homogenous_fips_mode',
+                metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS,
+                leader_node=leader) == 0,
+            timeout_sec=5,
+            backoff_sec=0.2,
+            err_msg="Metrics endpoint never returned 0 for initial check")
+
+        self.logger.debug(f'Restarting node {target_node} in non-FIPS mode')
+
+        self.redpanda.restart_nodes([target_node],
+                                    override_cfg_params=NON_FIPS_CONFIG)
+
+        wait_until(
+            lambda: get_metrics_value(
+                metrics_name='redpanda_cluster_non_homogenous_fips_mode',
+                metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS,
+                leader_node=leader) == 1,
+            timeout_sec=5,
+            backoff_sec=0.2,
+            err_msg=
+            f"Metrics endpoint never returned 1 after bringing up {target_node} in non-FIPS mode"
+        )
+
+        # We need to give the controller log a moment to update
+        sleep(5)
```

Comment:
question (knee-jerk reaction to `sleep`): what exactly are we waiting for? and, if it's observable, could we `wait_until` it occurs instead?

Reply:
Fair... the test was failing because I was restarting Redpanda too quickly for the controller log to update to capture the new node state.

Reply:
I'll investigate this some more tomorrow...

Reply:
Ya i figured as much. I don't have a sense of how generous 5s might be, so dealer's choice i guess.

Reply:
Ok got a fix up for it, no sleep required